examples/chef/common/clusters/channel/ChannelManager.cpp

```diff
@@ -0,0 +1,343 @@
+    pbs.majorNumber       = static_cast<uint8_t>(9);
+    pbs.minorNumber       = static_cast<uint16_t>(1);
+    mChannels.push_back(pbs);
+
+    ChannelInfoType pbsKids;
+    pbsKids.affiliateCallSign = MakeOptional(chip::CharSpan::fromCharString("KCTS"));
+    pbsKids.callSign          = MakeOptional(chip::CharSpan::fromCharString("KCTS-TV"));
+    pbsKids.name              = MakeOptional(chip::CharSpan::fromCharString("PBS Kids"));
+    pbsKids.majorNumber       = static_cast<uint8_t>(9);
+    pbsKids.minorNumber       = static_cast<uint16_t>(2);
+    mChannels.push_back(pbsKids);
+
+    ChannelInfoType worldChannel;
+    worldChannel.affiliateCallSign = MakeOptional(chip::CharSpan::fromCharString("KCTS"));
+    worldChannel.callSign          = MakeOptional(chip::CharSpan::fromCharString("KCTS-TV"));
+    worldChannel.name              = MakeOptional(chip::CharSpan::fromCharString("World Channel"));
+    worldChannel.majorNumber       = static_cast<uint8_t>(9);
+    worldChannel.minorNumber       = static_cast<uint16_t>(3);
+    mChannels.push_back(worldChannel);
+
+    mCurrentChannelIndex = 0;
+    mCurrentChannel      = mChannels[mCurrentChannelIndex];
+
+    ProgramType program1;
+    program1.identifier = chip::CharSpan::fromCharString("progid-abc1");
+    program1.channel    = abc;
+    program1.title      = chip::CharSpan::fromCharString("ABC Title1");
+    program1.subtitle   = MakeOptional(chip::CharSpan::fromCharString("My Program Subtitle1"));
+    program1.startTime  = 0;
+    program1.endTime    = 30 * 60;
+
+    mPrograms.push_back(program1);
+
+    ProgramType program_pbs1;
+    program_pbs1.identifier = chip::CharSpan::fromCharString("progid-pbs1");
+    program_pbs1.channel    = pbs;
+    program_pbs1.title      = chip::CharSpan::fromCharString("PBS Title1");
+    program_pbs1.subtitle   = MakeOptional(chip::CharSpan::fromCharString("My Program Subtitle1"));
+    program_pbs1.startTime  = 0;
+    program_pbs1.endTime    = 30 * 60;
+
+    mPrograms.push_back(program_pbs1);
+
+    ProgramType program2;
+    program2.identifier = chip::CharSpan::fromCharString("progid-abc2");
+    program2.channel    = abc;
+    program2.title      = chip::CharSpan::fromCharString("My Program Title2");
+    program2.subtitle   = MakeOptional(chip::CharSpan::fromCharString("My Program Subtitle2"));
+    program2.startTime  = 30 * 60;
+    program2.endTime    = 60 * 60;
+
+    mPrograms.push_back(program2);
+
+    ProgramType program3;
+    program3.identifier = chip::CharSpan::fromCharString("progid-abc3");
+    program3.channel    = abc;
+    program3.title      = chip::CharSpan::fromCharString("My Program Title3");
+    program3.subtitle   = MakeOptional(chip::CharSpan::fromCharString("My Program Subtitle3"));
+    program3.startTime  = 0;
+    program3.endTime    = 60 * 60;
+
+    mPrograms.push_back(program3);
+}
+
+CHIP_ERROR ChannelManager::HandleGetChannelList(AttributeValueEncoder & aEncoder)
+{
+    // TODO: Insert code here
```

Comment:
please remove more of these "insert code here" bits

Reply:
done.

Reply:
done.